packages/plugin/src/rules/no-unused-fields.ts

```diff
@@ -1,10 +1,39 @@
-import { GraphQLSchema, TypeInfo, visit, visitWithTypeInfo } from 'graphql';
+import { FieldDefinitionNode, GraphQLSchema, TypeInfo, visit, visitWithTypeInfo } from 'graphql';
+import { FromSchema } from 'json-schema-to-ts';
+import { GraphQLESTreeNode } from '../estree-converter/types.js';
 import { SiblingOperations } from '../siblings.js';
-import { GraphQLESLintRule } from '../types.js';
+import { GraphQLESLintRule, GraphQLESLintRuleListener } from '../types.js';
 import { requireGraphQLSchemaFromContext, requireSiblingsOperations } from '../utils.js';
 
 const RULE_ID = 'no-unused-fields';
 
+const schema = {
+  type: 'array',
+  maxItems: 1,
+  items: {
+    type: 'object',
+    additionalProperties: false,
+    properties: {
+      ignoredFieldSelectors: {
+        type: 'array',
+        uniqueItems: true,
+        minItems: 1,
+        description: [
+          'Fields that will be ignored and are allowed to be unused.',
+          '',
+          '> These fields are defined by ESLint [`selectors`](https://eslint.org/docs/developer-guide/selectors). Paste or drop code into the editor in [ASTExplorer](https://astexplorer.net) and inspect the generated AST to compose your selector.',
```

Comment:
The [`naming-convention` rule](https://github.com/dimaMachina/graphql-eslint/blob/master/packages/plugin/src/rules/naming-convention.ts) already makes use of ESLint selectors within configurations and it felt good to follow suit here. While this adds some clunkiness to the configuration (allowing an individual field looks like `"FieldDefinition[name.value=field]"` instead of "field"), there are benefits in that allowing fields with wildcards, based on their parent object, etc. are implicitly supported.

Reply:
I think it's fine to use selectors here instead of plain strings because we have the flexibility to choose parent name and parent pattern, see

```js
[
  '[parent.name.value=PageInfo][name.value=endCursor]',
  '[parent.name.value=PageInfo][name.value=startCursor]',
  '[parent.name.value=PageInfo][name.value=hasNextPage]',
  '[parent.name.value=PageInfo][name.value=hasPreviousPage]',
  '[parent.name.value=/Edge$/][name.value=cursor]',
  '[parent.name.value=/Connection$/][name.value=pageInfo]',
]
```

we can specify end's with `Edge` and `Connection`, @maciesielka maybe we should update the default options to include it there?